### PR TITLE
Add LETKF/GETKF yamls for new MPAS-JEDI test case

### LINKED ISSUE #124: Add LETKF yaml for 2024052700 MPAS case
I am planning to add a yaml file to run LETKF for the 2024052700 MPAS case. I will also likely rename the current `sonde_singeob_airTemperature_mpasjedi.yaml` to have separate EnVar and LETKF yaml files.

### PR DESCRIPTION
**Pull request by SamuelDegelia-NOAA**
This PR adds yaml files for LETKF and GETKF for the new 2024052700 MPAS-JEDI test case.  The original yaml file for this case has been renamed to `rrfs-test/testinput/rrfs_mpasjedi_2024052700_Ens3Dvar.yaml` to be consistent with the new files and our older test cases. The run scripts for this new case are also updated with a line that can be uncommented to run these EnKF yamls instead of EnVar. The only changes needed to run LETKF/GETKF on the user-side are to modify `run_mpasjedi.sh` and comment out the lines for EnVar and uncomment the lines for GETKF/LETKF. 

I will also work on updating the [wiki page](https://github.com/NOAA-EMC/RDASApp/wiki/A-quick-start-guide-for-running-the-MPASJEDI-2024052700-case-on-Jet-or-Hera) to add a section on how to run LETKF/GETKF for this case. 

Note: these yaml files use a single task to run LETKF/GETKF instead of running in split mode. This is fine for small tests, but split mode is recommended for using lots of obs. Testing is still ongoing for further optimization when running lots of obs or in split mode. 

New files
* `rrfs-test/testinput/rrfs_mpasjedi_2024052700_letkf.yaml`
* `rrfs-test/testinput/rrfs_mpasjedi_2024052700_getkf.yaml`

Closes #124

### CODE REVIEW

**Comment by SamuelDegelia-NOAA** [comment]
A new section 8 is added to the [wiki page](https://github.com/NOAA-EMC/RDASApp/wiki/A-Quick-Start-Guide-for-Running-the-MPASJEDI-2024052700-case-on-Jet-or-Hera) for the 20240527 MPAS-JEDI describing how to run LETKF/GETKF.

**Review by chunhuazhou — COMMENTED** [review]
@SamuelDegelia-NOAA Thank you for adding the new case for the LETKF and GETKF! These are very helpful!
I tested them on Jet, with the EnVar and LETKF running successfully. GETKF gave me errors related to logp for vertical localization, saysing "ABORT: Uknown vertical coordinate unit logp".

**Comment by SamuelDegelia-NOAA** [comment]
Thank you very much @chunhuazhou for reviewing, and sorry for the delay in getting back to you! I am unable to repeat this error when testing on Hera. Would you mind sharing your run directory on Jet so I could have a closer look?

**Comment by chunhuazhou** [comment]
@SamuelDegelia-NOAA Thank you for getting back to me and looking into this! My GETKF run directory is located at /lfs4/BMC/wrfruc/Chunhua.Zhou/RDASApp/expr/mpas_2024052700.getkf

**Comment by SamuelDegelia-NOAA** [comment]
@chunhuazhou I'm not really sure what is causing this issue. I copied your run directory and GETKF completes without issues. You can see that here: `/lfs5/BMC/wrfruc/Samuel.Degelia/EXPDIR/mpas_2024052700_chunhua`. Would you mind trying it again?

**Comment by chunhuazhou** [comment]
@SamuelDegelia-NOAA  Looks like I might have been using an older version of the code. I just tried your RDASApp and the GETKF ran successfully! Thank you for testing it on Jet, and again for preparing LETKF and GETKF for the new case! Do you happen to have a yaml file for testing real observations?

**Comment by SamuelDegelia-NOAA** [comment]
@chunhuazhou Awesome! An ongoing issue with both LETKF and GETKF is that they become very slow for realistic amounts of obs. I found that GETKF can be much faster if running in split observer/solver mode but I am still testing that functionality. I am doing these tests with the mesonet obs included with the new MPAS case (~500k total obs). 

If you want to see those yaml files, examples for GETKF are on Hera in `/scratch2/NCEPDEV/stmp3/Samuel.Degelia/mpas_newcase/rrfs-data_mpasjedi_2024052700_GETKF`.  In that directory are two yamls, one for running the observer and one for running the solver: 

```
mpasjedi_letkf_enkfvalidation_allobs_OBSERVER.yaml
mpasjedi_letkf_enkfvalidation_allobs_SOLVER.yaml
```

**Comment by chunhuazhou** [comment]
@SamuelDegelia-NOAA Thank you for sharing the yaml files! I was looking at the MPAS-JEDI tutorial practice from NCAR and they are doing similar thing by splitting the LETKF into 3 steps: observer, solver and diagoma. So having your example yaml files for different steps are really helpful! Thank you!

**Comment by SamuelDegelia-NOAA** [comment]
@chunhuazhou It is good to know that they are also doing a posterior diagoma step! My GETKF solver yaml has the posterior observer disabled because it normally runs for all the modulated members (12 members for each real member) using the halo obs distribution which adds nearly 10 min. So adding an extra step to just run the oma on the regular 30 members seems like the right move.

**Review by TingLei-NOAA — DISMISSED** [review]

**Comment by SamuelDegelia-NOAA** [comment]
@chunhuazhou Would you mind approving this PR if the tests you did are working? I would like to get it merged so that I can update the ctests to use the new mpas-jedi case (including these new letkf, getkf yamls). Thank you!!

**Review by chunhuazhou — DISMISSED** [review]
Looks good to me! Thank you for adding the new cases @SamuelDegelia-NOAA

File: rrfs-test/testinput/rrfs_mpasjedi_2024052700_getkf.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    fraction of retained variance: 0.850
+    lengthscale: 0.55
+    lengthscale units: logp
+  inflation:
+    rtps: 0.95
+    rtpp: 0.6
+    mult: 1.1
+
+output mean prior:
+  filename: ./bg.$Y-$M-$D_$h.$m.$s.nc
```
Comment (guoqing-noaa) [inline rrfs-test/testinput/rrfs_mpasjedi_2024052700_getkf.yaml]: @SamuelDegelia-NOAA 
Sorry for this picky comment. Are you okay to change "bg" to "bkg" to be consistent with the 3DEnVar situation?

Comment (SamuelDegelia-NOAA) [reply rrfs-test/testinput/rrfs_mpasjedi_2024052700_getkf.yaml]: Not at all, just changed it. Good idea.

**Comment by SamuelDegelia-NOAA** [comment]
Sorry, but @chunhuazhou or @TingLei-NOAA could you re-review pending the small change suggested by @guoqing-noaa? Thanks!

**Review by chunhuazhou — APPROVED** [review]

**Review by TingLei-NOAA — APPROVED** [review]

**Review by ShunLiu-NOAA — APPROVED** [review]

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA 
Sorry that I did not get time to review this PR earlier. 

I have a question: should LETKF update 30 members? I did not see your instructions on this aspect. 
Currently, the 30 members are linked to files under fix/
If we want to modify them, we will need to copy them over.

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa The yaml files here are only configured to update the ensemble mean which is calculated during the solver. The ensemble members are not updated due to: 

```
driver:
  save posterior ensemble: false
```

So it should not be a problem with the linked files. Whenever we get to some real cycling tests, we will want reenable this option.

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA Maybe we can update section 8 of the quick start guide to reflect this?
For these posterior ensembles, do we overwrite the original files and generate new files?

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa Good idea, I will update the wiki. And you can actually overwrite or generate new files based on how you define the `output` section of the yaml.

**Pull request merged**